### PR TITLE
[1LP][RFR] expression test for ansible service

### PR DESCRIPTION
**Pull request by digitronik**
Signed-off-by: Nikhil Dhandre <ndhandre@redhat.com>

## Purpose or Intent

- test for BZ-1628727 (ansible service button expression)
- extended usecase 

### PRT Run
{{pytest: cfme/tests/automate/custom_button/test_service_objects.py::test_custom_button_expression_ansible_service}}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 601f0b8d1596267efa0238f89dbe717cc4e7d984

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by digitronik** [comment]
Depends on #9368

**Comment by digitronik** [comment]
Please ignore 510 user-visibility failure. Its bug, which we already raised... I would like to go with failures.
user_visibility

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 668c5f50f618dddfb90b04fe7279e9d4c5f0e933

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit b889cc7e51c5c53464c3527b736ce4921afd4c52

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit f5931e4f0cf16a26319d3c62e115bf549e06f463

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 8d40273dddef3afaeaf9030d6f48c619a0f26368

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0e09f85e1a3031c4d3cf0ceeebbbfa7d9f31fb94

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/custom_button/test_service_objects.py [hunk 1]
```diff
@@ -683,3 +715,81 @@ def test_custom_button_unassigned_behavior_objs(
...
+            3. Navigate to object Detail page
+            4. Check: button should not enable/visible without tag
+            5. Check: button should enable/visible with tag
+
+    Bugzilla:
+        1628727
+        1509959
+        1513498
+        1755229
+    """
```
Comment (ganeshhubale) [inline cfme/tests/automate/custom_button/test_service_objects.py]: I can see this BZ(https://bugzilla.redhat.com/show_bug.cgi?id=1755229) is in Assigned state and test coverage is +. Can we add this BZ to coverage marker?

Comment (digitronik) [reply cfme/tests/automate/custom_button/test_service_objects.py]: @ganeshhubale we have separate test case for  `BZ-1755229`. It was creating problem to create test setup in `module scope` so just i added `TODO` comment and also include that bug in Bugzilla tag so i know this is affecting this test.

**Comment by digitronik** [comment]
> LGTM :)
> But need to check PRT failures because dependent PR is merged.

@ganeshhubale  this depend on this PR as well https://github.com/ManageIQ/integration_tests/pull/9476

File: cfme/tests/automate/custom_button/test_service_objects.py [hunk 2]
```diff
@@ -683,3 +715,81 @@ def test_custom_button_unassigned_behavior_objs(
...
+            )
+
+            if tag:
+                if expression == "enablement":
+                    assert custom_button_group.item_enabled(button.text)
+                else:  # visibility
+                    assert button.text in custom_button_group.items
+            else:
+                if expression == "enablement":
+                    # Note: SSUI still fallow enablement behaviour like 5.9. In latest version
```
Comment (lina-nikiforova) [inline cfme/tests/automate/custom_button/test_service_objects.py]: small typo `fallow` :)

**Comment by dajoRH** [comment]
I detected some fixture changes in commit e415d372bf6c539509f203c1df96c2fe7f360151

The *local* fixture **`service_button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *vis_enb_button_service*

The *local* fixture **`vis_enb_button_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**